subsys/bluetooth/scan.c

```diff
@@ -1346,3 +1460,52 @@ int bt_scan_params_set(struct bt_le_scan_param *scan_param)
 
 	return 0;
 }
+
+int bt_scan_attempts_filter_device_generic_add(const bt_addr_le_t *addr,
+					       bool force)
+{
+	int err = 0;
+	struct conn_attempts_filter *filter = &bt_scan.attempts_filter;
+
+	k_mutex_lock(&conn_attempts_mutex, K_FOREVER);
+
+	/* Check if device is already in the filter array. */
+	for (size_t i = 0; i < filter->count; i++) {
+		struct conn_attempts_device *device = &filter->device[i];
+
+		if (bt_addr_le_cmp(addr, &device->addr) == 0) {
+			LOG_DBG("Device is already in the filter arrary");
+			goto out;
+		}
+	}
+
+	if (filter->count >= ARRAY_SIZE(filter->device)) {
+		if (force) {
+			if (!filter->oldest_device) {
+				filter->oldest_device = filter->device;
+			}
+
+			attempts_filter_force_add(filter, addr);
+		} else {
+			LOG_ERR("No memory for the new device filter");
+			err = -ENOMEM;
+		}
+
+		goto out;
+	}
```

Comment:
Maybe we could use if... else here instead of goto. It would be cleaner.

Reply:
Done